src/app-modules/docker-compose

```diff
@@ -166,73 +206,22 @@ RC_COMPOSITION_NOT_RUNNING=4
 RC_TIMEOUT_STARTING=8
 RC_MAX_ITERATIONS_EXHAUSTED=16
 
-wait_for_running() {
-    local i
-    local out_tmp
-    local all_running
-    local count
-    local expected_count
-    local rc="$RC_TIMEOUT_STARTING"
-    local max_iterations=32
-    local manifests_dir="$1"
-
-    out_tmp=$(mktemp)
-    all_running="false"
-    i=1
-    while test "$all_running" = "false"; do
-        sleep "$RUNNING_STATE_TIMEOUT_S"
-        $docker_compose_cmd --project-directory "$manifests_dir" ps -q > "$out_tmp"
-        count=$(cat "$out_tmp" | wc -l)
-        expected_count=$(docker-compose --project-directory "$manifests_dir" config --images | wc -l)
-        if test $count -ge $expected_count; then
-            cat "$out_tmp" | xargs -n1 -I{} docker inspect {} | jq -r '.[] | "\(.State.Running) \(.State.Status)"' | grep -qv 'true running'
-            if test $? -eq 1; then
-                $docker_compose_cmd --project-directory "$manifests_dir" ps -q | xargs -n1 -I{} docker inspect {} | jq -r '.[] | "\(.State.Running) \(.State.Status)"'
-                docker ps
-                rc=0
-                break
-            fi
-        fi
-        i=$(expr $i + 1)
-        if test "$i" -gt "$max_iterations"; then
-            rc="$RC_MAX_ITERATIONS_EXHAUSTED"
-            break
-        fi
-    done
-    rm -f "${out_tmp}"
-    return "$rc"
-}
-
-rc_to_message() {
-    local rc="$1"
-
-    if test "$rc" = "$RC_MAX_ITERATIONS_EXHAUSTED"; then
-        echo "max iterations waiting for composition to start exhausted"
-    fi
-    if test "$rc" = "$RC_COMPOSITION_NOT_RUNNING"; then
-        echo "docker compose process exited unexpectedly"
-    fi
-    if test "$rc" = "$RC_TIMEOUT_STARTING"; then
-        echo "time out reached during startup"
-    fi
-}
-
 app_rollout() {
     local application_name="$1"
     local manifests_dir="$2"
-    local pid
     local rc
-    local message
 
-    $docker_compose_cmd --project-directory "$manifests_dir" up -d > "$manifests_dir"/compose.log 2>&1 &
-    pid=$!
-    wait_for_running "$manifests_dir"
+    docker_compose_cmd \
+        --project-directory "$manifests_dir" \
+        --project-name "$application_name" \
+        up -d --wait --remove-orphans > "$manifests_dir"/compose.log 2>&1
     rc="$?"
     if test "$rc" != "0"; then
         echo "app_rollout failed to start composition, logs follow:" 1>&2
-        $docker_compose_cmd --project-directory "$manifests_dir" logs -t 1>&2
-        message=$(rc_to_message "$rc")
-        echo "app_rollout failed due to: $message." 1>&2
+        docker_compose_cmd \
+            --project-directory "$manifests_dir" \
+            --project-name "$application_name" \
+            logs -t 1>&2
         return $rc
```

Comment:
I know it was not in the original code, but 
```
cat "$manifests_dir"/compose.log
```
would be nice. remember last test failure? the log was empty, because the test simulated lack of images/commands.

Reply:
Do we need the `compose.log` file? The output from docker compose should be contained in the client deployment logs now that the command is not running in the background.